src/runner.ts

```diff
@@ -202,6 +208,124 @@ export class ActionRunner {
+          const members = await this.teamApi.getTeamMembers(team);
+          usersToAdd = [...new Set([...usersToAdd, ...members])];
+        }
+      }
+      requirements.push({ users: usersToAdd, requiredApprovals: reviewers.min_approvals });
+    }
+
+    // We count how many reviews are needed in total
+    const requiredAmountOfReviews = rule.reviewers.map((r) => r.min_approvals).reduce((a, b) => a + b, 0);
+    // We get the list of users that approved the PR
+    const approvals = await this.prApi.listApprovedReviewsAuthors();
+
+    // Utility method used to generate error
+    const generateErrorReport = (): ReviewReport => {
+      const filterMissingUsers = (reviewData: { users?: string[] }[]): string[] =>
+        reviewData.flatMap((r) => r.users ?? []).filter((u) => approvals.indexOf(u) < 0);
+
+      // Calculating all the possible combinations to see the missing reviewers is very complicated
+      // Instead we request everyone who hasn't reviewed yet
+      return {
+        missingReviews: requiredAmountOfReviews,
+        missingUsers: filterMissingUsers(requirements),
+        teamsToRequest: rule.reviewers.flatMap((r) => r.teams ?? []),
+        usersToRequest: filterMissingUsers(rule.reviewers),
+      };
+    };
+
+    // If not enough reviews (or no reviews at all)
+    if (approvals.length < requiredAmountOfReviews) {
+      this.logger.warn(`Not enough approvals. Need at least ${requiredAmountOfReviews} and got ${approvals.length}`);
+      // We return an error and request reviewers
+      return [false, generateErrorReport()];
+    }
+
+    this.logger.debug(`Required users to review: ${JSON.stringify(requirements)}`);
+
+    const conditionApprovals: {
+      matchingUsers: string[];
+      requiredUsers: string[];
+      requiredApprovals: number;
+    }[] = [];
+
+    // Now we see, from all the approvals, which approvals could match each rule
+    for (const { users, requiredApprovals } of requirements) {
+      const ruleApprovals = approvals.filter((ap) => users.indexOf(ap) !== -1);
+      conditionApprovals.push({ matchingUsers: ruleApprovals, requiredUsers: users, requiredApprovals });
+    }
+    this.logger.debug(`Matching approvals: ${JSON.stringify(conditionApprovals)}`);
+
+    // If one of the rules doesn't have the required approval we fail the evaluation
+    if (conditionApprovals.some((cond) => cond.matchingUsers.length === 0)) {
+      this.logger.warn("One of the groups does not have any approvals");
+      return [false, generateErrorReport()];
+    }
+
+    /**
+     * We split all the reviewers that have more than one required review into its own object
+     * So if a there is a [requiredApprovals: 2, users: ["abc", "def"]]
+     * It is split into [requiredApprovals: 1, users: ["abc", "def"]], [requiredApprovals: 1, users: ["abc", "def"]]
+     * This allows us to be more flexible when testing cases
+     */
+    const splittedRequirements = conditionApprovals.flatMap((reviewer) =>
```

Comment:
`splittedRequirements` is not actually used (other than in a logger log) :D

Here is test that is failing because of that:

```
test("should fail with this setup", async () => {
  const rule: AndDistinctRule = {
    type: RuleTypes.AndDistinct,
    reviewers: [
      { users: ["user-1", "user-2"], min_approvals: 2 },
      { users: ["user-1"], min_approvals: 1 },
    ],
    name: "test",
    condition: { include: [] },
  };
  api.listApprovedReviewsAuthors.mockResolvedValue([...users]);
  const [result] = await runner.andDistinctEvaluation(rule);
  expect(result).toBe(false);
});
```

Reply:
Great catch. Thanks for that! I missplaced it in a refactoring